support-workers/src/main/scala/com/gu/emailservices/ContributionEmailFields.scala

```diff
@@ -3,56 +3,56 @@ package com.gu.emailservices
 import com.gu.emailservices.SubscriptionEmailFieldHelpers.{formatDate, hyphenate, mask}
 import com.gu.i18n.Currency
 import com.gu.salesforce.Salesforce.SfContactId
-import com.gu.support.redemptions.RedemptionData
 import com.gu.support.workers._
 import org.joda.time.DateTime
 
 case class ContributionEmailFields(
-    email: String,
-    created: DateTime,
-    amount: BigDecimal,
-    currency: Currency,
-    edition: String,
-    name: String,
+  created: DateTime,
+  amount: BigDecimal,
+  paymentMethod: PaymentMethod
+) extends AllProductsEmailFields {
+
+  def apply(
     billingPeriod: BillingPeriod,
+    user: User,
+    currency: Currency,
     sfContactId: SfContactId,
-    paymentMethod: Either[PaymentMethod, RedemptionData],
-    directDebitMandateId: Option[String] = None
-) extends EmailFields {
+    directDebitMandateId: Option[String],
+  ): EmailFields = new EmailFields {
 
-  val paymentFields = paymentMethod match {
-    case Left(dd: DirectDebitPaymentMethod) => List(
-      "account name" -> dd.bankTransferAccountName,
-      "account number" -> mask(dd.bankTransferAccountNumber),
-      "sort code" -> hyphenate(dd.bankCode),
-      "Mandate ID" -> directDebitMandateId.getOrElse(""),
-      "first payment date" -> formatDate(created.plusDays(10).toLocalDate),
-      "payment method" -> "Direct Debit"
-    )
-    case Left(dd: ClonedDirectDebitPaymentMethod) => List(
-      "account name" -> dd.bankTransferAccountName,
-      "account number" -> mask(dd.bankTransferAccountNumber),
-      "sort code" -> hyphenate(dd.bankCode),
-      "Mandate ID" -> dd.mandateId,
-      "first payment date" -> formatDate(created.plusDays(10).toLocalDate),
-      "payment method" -> "Direct Debit"
-    )
-    case Left(_: PayPalReferenceTransaction) => List("payment method" -> "PayPal")
-    case Left(_: CreditCardReferenceTransaction) => List("payment method" -> "credit / debit card")
-    case Right(_: RedemptionData) => Nil
```

Comment:
this was an invalid state as there's no such thing as a redeemable contribution, so this will fail the lambda earlier and cause a support worker failure now